minions/async/src/main/java/io/github/microcks/minion/async/producer/ProducerManager.java

```diff
@@ -153,36 +153,19 @@ private String renderEventMessageContent(EventMessage eventMessage) {
       return content;
    }
 
-   /** Get the Kafka topic name corresponding to a AsyncMockDefinition, sanitizing all parameters. */
-   private String getKafkaTopicName(AsyncMockDefinition definition) {
-      // Produce service name part of topic name.
-      String serviceName = definition.getOwnerService().getName().replace(" ", "");
-      serviceName = serviceName.replace("-", "");
-      // Produce version name part of topic name.
-      String versionName = definition.getOwnerService().getVersion().replace(" " , "");
-      // Produce operation name part of topic name.
-      String operationName = definition.getOperation().getName();
-      if (operationName.startsWith("SUBSCRIBE ")) {
-         operationName = operationName.substring(operationName.indexOf(" ") + 1);
+   public static String replacePartPlaceholders(EventMessage eventMessage, String operationName) {
+      String partsCriteria = eventMessage.getDispatchCriteria();
+      if (partsCriteria != null && !partsCriteria.isBlank()) {
+         String[] criterion = partsCriteria.split("/");
+         for (String criteria : criterion) {
+            if (criteria != null && !criteria.isBlank()) {
+               String[] element = criteria.split("=");
+               String key = String.format("\\{%s\\}", element[0]);
+               operationName = operationName.replaceAll(key, URLEncoder.encode(element[1], Charset.defaultCharset()));
+            }
+         }
       }
-      operationName = operationName.replace('/', '-');
-      // Aggregate the 3 parts using '-' as delimiter.
-      return serviceName + "-" + versionName + "-" + operationName;
+      return operationName;
```

Comment:
Can we check here if we could reuse something from `io.github.microcks.util.URIBuilder` ?